agent/agent.go

```diff
@@ -190,11 +190,12 @@ func (a *agent) fetch() ([]byte, error) {
 		return nil, err
 	}
 	q := hex.EncodeToString(id)
-	url := fmt.Sprintf("%s/%s", a.svcURL, q)
+	url := fmt.Sprintf("%s/%s", a.svcURL, "")
```

Comment:
No need for this line (also, this will result in `/` at the end of the URL), you can remove this line and use `http.NewRequest(http.MethodGet, a.svcURL, nil)`.